src/direction.rs

```diff
@@ -200,6 +200,73 @@ impl Direction {
         Self::BottomRight,
     ];
 
+    /// Iterates through all enum variant in order
+    pub fn iter() -> impl Iterator<Item = Self> {
+        Self::ALL_DIRECTIONS.into_iter()
+    }
+
+    #[inline]
+    #[must_use]
+    #[doc(alias = "clockwise")]
+    /// Returns the next direction in clockwise order
+    pub const fn right(self) -> Self {
+        match self {
+            Self::TopRight => Self::BottomRight,
+            Self::Top => Self::TopRight,
+            Self::TopLeft => Self::Top,
+            Self::BottomLeft => Self::TopLeft,
+            Self::Bottom => Self::BottomLeft,
+            Self::BottomRight => Self::Bottom,
+        }
+    }
+
+    #[inline]
+    #[must_use]
+    #[doc(alias = "counterclockwise")]
+    /// Returns the next direction in counter clockwise order
+    pub const fn left(self) -> Self {
+        match self {
+            Self::TopRight => Self::Top,
+            Self::Top => Self::TopLeft,
+            Self::TopLeft => Self::BottomLeft,
+            Self::BottomLeft => Self::Bottom,
+            Self::Bottom => Self::BottomRight,
+            Self::BottomRight => Self::TopRight,
+        }
+    }
+
+    #[inline]
+    #[must_use]
+    /// Rotates `self` counter clockwise by `mid` amount.
```

Comment:
The term "mid" doesn't mean anything to me. I would maybe call this `offset` or `rotation`?